gadget/install/content.go

```diff
@@ -110,5 +114,22 @@ func writeFilesystemContent(laidOut *gadget.LaidOutStructure, fsDevice string, o
 		return fmt.Errorf("cannot create filesystem image: %v", err)
 	}
 
+	// For data partition, build drivers tree if required, so kernel
+	// drivers are available on first boot of the installed system.
+	if laidOut.Role() == gadget.SystemData && kSnapInfo != nil && kSnapInfo.NeedsDriversTree {
+		destRoot := mountpoint
+		if kSnapInfo.IsCore {
+			destRoot = filepath.Join(mountpoint, "system-data")
+		}
+		destDir := filepath.Join(dirs.SnapKernelDriversTreesDirUnder(destRoot),
+			kSnapInfo.Name, kSnapInfo.Revision.String())
+		logger.Noticef("building drivers tree in %s", destDir)
+
+		if err := kernelEnsureKernelDriversTree(kSnapInfo.MountPoint, destDir, nil,
```

Comment:
the new signature feels a bit clunky?  maybe we could make KernelSnapInfo implement one of the snap package interfaces and go go back to something a bit cleaner?